docs/hoon/library/1b.md

```diff
@@ -1,147 +1,294 @@
----
-navhome: /docs/
----
+  ---
+  navhome: /docs/
+  ---
 
 
-# 1b: tree addressing
-### `++cap`
+  # 1b: tree addressing
 
-Tree head
+  Check out the Nock [explanation](docs/nock/explanation/) for more
+  information on the tree-addressing system.
 
-Tests whether an `a` is in the head or tail of a noun. Produces the
-cube `%2` if it is within the head, or the cube `%3` if it is
-within the tail.
+  ### `++cap`
 
-Accepts
--------
+  Tree head
 
-`a` is an atom.
+  Tests whether the tree address `a` is in the head or the tail of a noun.
+  Produces the constant atom `%2` if it is within the head (subtree `+2`), or
+  the constant atom `%3` if it is within the tail (subtree `+3`).
 
-Produces
---------
+  In the Urbit binary-tree system, a parent at address `n` will always have its
+  children at address `2n` (the head) and address `2n+1` (the tail).
 
-A cube.
 
-Source
-------
+  Accepts
+  -------
 
-        ++  cap                                                 ::  tree head
-          ~/  %cap
-          |=  a=@
-          ^-  ?(%2 %3)
-          ?-  a
-            %2        %2
-            %3        %3
-            ?(%0 %1)  !!
-            *         $(a (div a 2))
-          ==
-        ::
+  `a` is an atom.
 
-Examples
---------
+  Produces
+  --------
 
-    > (cap 4)
-    %2
-    > (cap 6)
-    %3
-    > (cap (add 10 9))
-    %2
+  A constant atom.
 
+  Source
+  ------
 
+          ++  cap                                                 ::  tree head
+            ~/  %cap
+            |=  a=@
+            ^-  ?($2 $3)
+            ?-  a
+              $2        %2
+              $3        %3
+              ?(%0 %1)  !!
+              *         $(a (div a 2))
+            ==
+          ::
 
-***
-### `++mas`
+  Examples
+  --------
 
-Axis within head/tail?
+      > (cap 4)
+      %2
 
-Computes the axis of `a` within either the head or tail of a noun (depends whether `a` lies within the the head or tail).
+      > (cap 6)
+      %3
 
-Accepts
--------
+      > (cap (add 10 9))
+      %2
 
-`a` is an [atom]().
+      > (cap 1)                                    ::axis '0' is in neither the head nor the tail
+      ! exit
 
-Produces
---------
+      > (cap 0x40))                                ::0x1a is of the hexadecimal soft atom type (@ux)
+      %2
+      > `@`0x40
+      64                           
 
-An atom.
+      > (cap 'a')                                  ::'a' is of the cord (text) soft atom type (@t)
+      %3
+      > `@`'a'
+      97                                        
 
-Source
-------
 
-        ++  mas                                                 ::  tree body
-          ~/  %mas
-          |=  a=@
-          ^-  @
-          ?-  a
-            1   !!
-            2   1
-            3   1
-            *   (add (mod a 2) (mul $(a (div a 2)) 2))
-          ==
-        ::
+  ***
+  ### `++mas`
 
-Examples
---------
+  Address within head/tail?
```

Comment:
[I don't like the question mark here](https://en.wikipedia.org/wiki/High_rising_terminal).